src/os.rs

```diff
@@ -14,55 +14,53 @@
 use std::fmt;
 use rand_core::{RngCore, Error, impls};
 
-/// A random number generator that retrieves randomness straight from
-/// the operating system.
+/// A random number generator that retrieves randomness straight from the
+/// operating system. This is the preferred external source of randomness for
+/// most applications. Commonly it is used to initialize a user-space RNG, which
```

Comment:
I still don't think the guidance about what to use this for / which RNG is 'better' belongs here at all. (I'm thinking of moving the doc about crypto-stuff from lib.rs to the README; we can reference that.)

The note about `EntropyRng` and details like not blocking and how this interfaces with the OS make sense.

I think we should also add a note that `OsRng::new()` is guaranteed to be very cheap (after first successful call), and will never consume more than one file handle per process.

Reply:
Removed the paragraph and added a bit about `OsRng::new()`.

Reply:
Also the bit about *this is the preferred source of randomness* — I presume you mean entropy source, but it could also be read to mean preferred way to get each new random value.